In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data=pd.read_csv('E:\Excelr files\data\Hypothesis testing\BuyerRatio.csv')

In [3]:
data.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Observed Values  2 non-null      object
 1   East             2 non-null      int64 
 2   West             2 non-null      int64 
 3   North            2 non-null      int64 
 4   South            2 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 208.0+ bytes


α =0.05 significance level
x=discrete
y=discrete

There is more than 2 variable we perform chi2 test

In [ ]:
#Null hypothesis(H0)=The male-female buyerRatio is similar across region.
#Alternate hypothesis(Ha)=The male-female buyer ratio are different across region

In [5]:
stats.chi2_contingency([data["East"], data["West"], data["North"], data["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

P-value of chi2 test =0.66 > α 
Thus Ho is accepted.
The male-female buyer ratio is similar across the region.